src/utils/crypto.ts

```diff
@@ -184,22 +197,22 @@ export function sign (data, privateKey) {
  * @param {Buffer} publicKey - Key to verify against
  * @return {Boolean} Valid?
  */
-export function verify (str, signature, publicKey) {
-  return nacl.sign.detached.verify(new Uint8Array(str), signature, publicKey)
+export function verify (str: string|Buffer, signature: Buffer, publicKey: Buffer): boolean {
+  return nacl.sign.detached.verify(new Uint8Array(<Buffer>str), signature, publicKey)
```

Comment:
```suggestion
export function verify (
  data: Buffer, signature: Buffer | Uint8Array, publicKey: Buffer): boolean {
  return nacl.sign.detached.verify(new Uint8Array(data), signature, publicKey)
}
```

The main reason I suggest removing casting and renaming str to data and change its type to Buffer is 
1. Throughout the sdk, the first parameter of `verify` is always passed as a Buffer
2. `Uint8Array` does not accept primitive `string` but `Buffer` instead

(JSDoc comments can be updated as well in this case)

Reply:
Second parameter is suggested to have a type of `Buffer | Uint8Array` in order to remove casting